### PR TITLE
[WIP] BR: do not filter non-system tables when using --with-sys-table

### PR DESCRIPTION
**Pull request by IANTHEREAL**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/restore.go [hunk 1]
```diff
@@ -486,7 +486,7 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 	// to avoid import cycle problem which we won't do it in this pr, then refactor this
 	//
 	// if it's point restore and reached here, then cmdName=FullRestoreCmd and len(cfg.FullBackupStorage) > 0
-	if cmdName == FullRestoreCmd && cfg.WithSysTable {
```
Comment (D3Hunter) [inline br/pkg/task/restore.go:489]: cannot remove this, if we `InitFullClusterRestore` we will check whether cluster is fresh & whether sys table compatible

File: br/pkg/restore/systable_restore.go [hunk 2]
```diff
@@ -96,6 +101,16 @@ func (rc *Client) RestoreSystemSchemas(ctx context.Context, f filter.Filter) {
 	tablesRestored := make([]string, 0, len(originDatabase.Tables))
 	for _, table := range originDatabase.Tables {
 		tableName := table.Info.Name
+		// --with-sys-table must be specified for recoverable system tables.
+		// One drawback of this filtering algorithm is that if a system table is not in both the unrecoverable and recoverable table list,
+		// the system table can be recovered  using only `--filter` flag
+		if !rc.withSysTable && isRecoverableSysTable(tableName.L) {
```
Comment (D3Hunter) [inline br/pkg/restore/systable_restore.go:107]: this list only contains privileges table, there're other systables

File: br/pkg/task/restore.go (hunk 1 above)
Comment (IANTHEREAL) [reply br/pkg/task/restore.go:489]: I don't learn all code, but it doesn't need  `cfg.WithSysTable` here before https://github.com/pingcap/tidb/pull/36652/files#diff-7623fc7b2d82b31b389e3575c938daf9f5724962391d8cb21968b541c45c3191R489

File: br/pkg/restore/systable_restore.go (hunk 2 above)
Comment (IANTHEREAL) [reply br/pkg/restore/systable_restore.go:107]: other systables are in `unRecoverableTable` list, is it ok?

File: br/pkg/task/restore.go (hunk 1 above)
Comment (IANTHEREAL) [reply br/pkg/task/restore.go:489]: I don't understand these code clearly, should  the `rc.fullClusterRestore`  be `!explicitFilter && rc.IsFull() && !cfg.WithSysTable`? Only restoring systbale, br need to check ifcluster is fresh and sys table compatiblable

**Pull request closed without merging**